src/components/Modal/Modal.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import Modal from './Modal';
+
+describe('Modal', () => {
+  const onCloseMock = vi.fn();
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('renders the modal when isOpen is true', () => {
+    render(
+      <Modal isOpen={true} onClose={onCloseMock}>
+        <div>Modal Content</div>
+      </Modal>,
+    );
+
+    expect(screen.getByTestId('modal-button')).toBeInTheDocument();
+    expect(screen.getByLabelText('Close Modal')).toBeInTheDocument();
+    expect(screen.getByText('Modal Content')).toBeInTheDocument();
+  });
+
+  it('does not render the modal when isOpen is false', () => {
+    render(
+      <Modal isOpen={false} onClose={onCloseMock}>
+        <div>Modal Content</div>
+      </Modal>,
+    );
+
+    expect(screen.queryByTestId('modal-button')).not.toBeInTheDocument();
+    expect(screen.queryByLabelText('Close Modal')).not.toBeInTheDocument();
+    expect(screen.queryByText('Modal Content')).not.toBeInTheDocument();
+  });
+
+  it('calls the onClose function when the close button is clicked', () => {
+    render(
+      <Modal isOpen={true} onClose={onCloseMock}>
+        <div>Modal Content</div>
+      </Modal>,
+    );
+
+    const closeButton = screen.getByLabelText('Close Modal');
+    fireEvent.click(closeButton);
```

Comment:
minor nitpick since it has no difference here but should we standardize on using `user-event` over `fireEvent`?